lib/provider/immoscout-mobile.js

```diff
@@ -0,0 +1,161 @@
+import utils, {buildHash} from '../utils.js';
+import queryString from 'query-string';
+let appliedBlackList = [];
+
+async function getListings(url) {
+  const response = await fetch(url, {
+    method: 'POST',
+    headers: {
+      'User-Agent': 'ImmoScout24_1410_30_._',
+      'Content-Type': 'application/json',
+    },
+    body: JSON.stringify({
+      supportedResultListTypes: [],
+      userData: {}
+    })
+  });
+  if (!response.ok) {
+    console.error('Error fetching data from ImmoScout Mobile API:', response.statusText);
+    return [];
+  }
+
+  const responseBody = await response.json();
+  return responseBody.resultListItems.filter((item) => item.type === 'EXPOSE_RESULT').map(expose => {
+    const item = expose.item;
+    const [price, size, ] = item.attributes;
+    return {
+      id: item.id,
+      price: price?.value,
+      size: size?.value,
+      title: item.title,
+      link: `${metaInformation.baseUrl}/expose/${item.id}`,
+      address: item.address?.line,
+    };
+  });
+}
+
+function nullOrEmpty(val) {
+  return val == null || val.length === 0;
+}
+function normalize(o) {
```

Comment:
please run the linter, it will most likely find some flaws in here :)

Reply:
I've actually run `npm lint`, but nothing is that file is flagged.

Reply:
interesting.
`yarn lint` finds some errors (funny enough none of which caused by you)
als I'd suggest to run `yarn run format`.

Actually there should be a pre-commit hook...